app/__test__/attribute/attribute.schema.test.ts

```diff
@@ -0,0 +1,147 @@
+import { CreateAttributeInput, createAttributeSchema } from "@/app/(backend)/api/attribute/attribute.schema";
+
+it("should validate input with all required fields", () => {
+  const input = {
+    size: "small",
+    color: "#FF0000",
+    quantity: 10,
+    image: "https://example.com/image.jpg",
+  };
+
+  const result = createAttributeSchema.parse(input);
+
+  expect(result).toEqual(input);
+});
+
+it("should trim string fields", () => {
+  const input = {
+    size: "   small   ",
+    color: "   #FF0000   ",
+    quantity: 10,
+    image: "   https://example.com/image.jpg   ",
+  };
+
+  const expected = {
+    size: "small",
+    color: "#FF0000",
+    quantity: 10,
+    image: "https://example.com/image.jpg",
+  };
+
+  const result = createAttributeSchema.parse(input);
+
+  expect(result).toEqual(expected);
+});
+
+it("should throw error when size field is missing", () => {
+  const input = {
+    color: "#FF0000",
+    quantity: 10,
+    image: "https://example.com/image.jpg",
+  };
+
+  expect(() => {
+    createAttributeSchema.parse(input);
+  }).toThrow("The size is mandatory");
+});
+
+it("should throw error when color field is missing", () => {
+  const input = {
+    size: "small",
+    quantity: 10,
+    image: "https://example.com/image.jpg",
+  };
+
+  expect(() => {
+    createAttributeSchema.parse(input);
+  }).toThrow("The color is mandatory");
+});
+
+it("should throw error when quantity field is missing", () => {
+  const input = {
+    size: "small",
+    color: "#FF0000",
+    image: "https://example.com/image.jpg",
+  };
+
+  expect(() => {
+    createAttributeSchema.parse(input);
+  }).toThrow("The quantity is mandatory");
+});
+
+it("should throw error when image field is missing", () => {
+  const input = {
+    size: "small",
+    color: "#FF0000",
+    quantity: 10,
+  };
+
+  expect(() => {
+    createAttributeSchema.parse(input);
+  }).toThrow("The image URL is mandatory");
+});
+
+it("should validate input with a valid hex color", () => {
+  const input = {
+    size: "small",
+    color: "#FF0000",
+    quantity: 10,
+    image: "https://example.com/image.jpg",
+  };
+
+  const result = createAttributeSchema.parse(input);
+
+  expect(result).toEqual(input);
+});
+
+it("should throw an error when the color field is not a valid hex color", () => {
+  const input = {
+    size: "small",
+    color: "invalid",
+    quantity: 10,
+    image: "https://example.com/image.jpg",
+  };
+
+  expect(() => createAttributeSchema.parse(input)).toThrow(
```

Comment:
`toThrowError`

Reply:
![image](https://github.com/Los-creativos/cloteck-e/assets/90226510/c3de224b-aa17-41d3-8af7-abf8e33519f1)

In jest 29.7.0 it seems to be deprecated